backend/service/src/notification_config/parameters.rs

```diff
@@ -0,0 +1,80 @@
+use std::collections::HashMap;
+
+use crate::service_provider::ServiceContext;
+use crate::notification::NotificationServiceError;
+use super::query::NotificationConfig;
+use repository::NotificationQueryRowRepository;
+
+pub fn get_notification_parameters(
+    ctx: &ServiceContext,
+    notification_config: &NotificationConfig,
+) -> Result<Vec<HashMap<String, serde_json::Value>>, NotificationServiceError> {
+    // Fetch default parameters from config
+    let params_string = match notification_config.parameters.len() {
+        0 => "[]".to_string(),
+        _ => notification_config.parameters.clone(),
+    };
+    let sql_params_string = match &notification_config.parameter_query_id {
+        None => "[]".to_string(),
+        Some(query_id) => get_sql_parameters(ctx, query_id)?,
+    };
+
+    // Parse both sets of parameter strings, then merge the resulting vectors
+    let mut all_params: Vec<HashMap<String, serde_json::Value>> = serde_json::from_str(&params_string)
+        .map_err(|e| {
+            NotificationServiceError::InternalError(format!(
+                "Failed to parse notification config parameters (expecting an array of params_string): {:?} - {}",
+                e, params_string
+            ))
+        })?;
+    let mut sql_params: Vec<HashMap<String, serde_json::Value>>  = serde_json::from_str(&sql_params_string)
+        .map_err(|e| {
+            NotificationServiceError::InternalError(format!(
+                "Failed to parse notification sql parameters (expecting an array of params_string): {:?} - {}",
+                e, params_string
+            ))
+        })?;
+    all_params.append(&mut sql_params);
```

Comment:
This can be written like this instead.
I'm not sure if one is strictly better than the other but in general I try to only make this mutable if they have to be.
```suggestion
    let sql_params: Vec<HashMap<String, serde_json::Value>>  = serde_json::from_str(&sql_params_string)
        .map_err(|e| {
            NotificationServiceError::InternalError(format!(
                "Failed to parse notification sql parameters (expecting an array of params_string): {:?} - {}",
                e, params_string
            ))
        })?;
    all_params.extend(sql_params);
```
Weird to me that append requires `&mut`, but I guess it needs the same datatype...

I think it's fine how it is but wanted to check why i needed to be `mut`